src/ng/animate.js

```diff
@@ -416,8 +416,8 @@ var $AnimateProvider = ['$provide', /** @this */ function($provide) {
        * @param {Promise} animationPromise The animation promise that is returned when an animation is started.
        */
       cancel: function(runner) {
-        if (runner.end) {
-          runner.end();
+        if (runner.cancel) {
+          runner.cancel();
```

Comment:
Right. I think we have two problems (and my suggestion only takes care of one of them). So, in addition we should also change the cancellation code as follows:

```js
// Before:
for (i = 0, ii = previousLeaveAnimations.length; i < ii; ++i) {
  $animate.cancel(previousLeaveAnimations[i]);
}
previousLeaveAnimations.length = 0;

// After:
while (previousLeaveAnimations.length) {
  $animate.cancel(previousLeaveAnimations.shift());
}
```

--
In any case, I think it is fine to traverse the array backwards as an alternative (if it doesn't create other problems).